Documentation/devel/release.md

```diff
@@ -1,9 +1,22 @@
 # rkt release guide
 
-How to perform a release of rkt.
-This guide is probably unnecessarily verbose, so improvements welcomed.
-Only parts of the procedure are automated; this is somewhat intentional (manual steps for sanity checking) but it can probably be further scripted, please help.
+## Release cycle
+
+This section describes the typical release cycle of rkt:
+
+1. A [milestone](https://github.com/coreos/rkt/milestones) is created on GitHub, fixing the target date for a future release (typically every 2/3 weeks).
+2. Tickets milestoned for the next release are worked on, accordingly to their priorities.
+3. PRs are submitted in order to introduce changes. Those needs to be reviewed and tests must not be failing in order to be accepted.
```

Comment:
`3. Changes are submitted for review in the form of a GitHub Pull Request (PR). Each PR undergoes review and must pass continuous integration (CI) tests before being accepted and merged into the main line of rkt source code.`